src/client-side-encryption/state_machine.ts

```
@@ -218,14 +220,15 @@ export class StateMachine {
             );
           }
 
-          const collInfo = await this.fetchCollectionInfo(
+          const collInfoCursor = this.fetchCollectionInfo(
             metaDataClient,
             context.ns,
             filter,
             options
           );
-          if (collInfo) {
-            context.addMongoOperationResponse(collInfo);
+
+          for await (const collInfo of collInfoCursor) {
+            context.addMongoOperationResponse(serialize(collInfo));
```

`addMongoOperationResponse` squashes errors (oops).  

Can we add error handling, similar to C's approach?  This will require bindings changes.

https://github.com/kevinAlbs/mongo-c-driver/commit/eb18fc82e822424a4a84c7754debd707ba3b1e9a

---

What bindings changes are needed? Can the `context.state` be checked after each call here?

---

I suppose, but I think that goes against our existing patterns in mongodb-client-encryption.  

1. We don't provide a wrapper around `mongocrypt_status_message`, so we wouldn't be able to throw the error message that libmongocrypt generates.
2. Most (all?) other bindings functions that call libmongocrypt functions that can error handle the error inline and throw an error if we encounter one, instead of returning nothing and relying on the state machine to handle it. 

Ex:
```c++
    if (!mongocrypt_ctx_encrypt_init(context.get(), ns.c_str(), ns.size(), binaryCommand.get())) {
        throw TypeError::New(Env(), errorStringFromStatus(context.get()));
    }
```

so I'm imagining something like:
```c++
void MongoCryptContext::AddMongoOperationResponse(const CallbackInfo& info) {
    Uint8Array buffer = Uint8ArrayFromValue(info[0], "buffer");

    std::unique_ptr<mongocrypt_binary_t, MongoCryptBinaryDeleter> reply_bson(
        Uint8ArrayToBinary(buffer));
    if (!mongocrypt_ctx_mongo_feed(context(), reply_bson.get())) {
        throw Error::New(Env(), errorStringFromStatus(context()));
    }
}
```

---

Why isn't this problematic for existing driver releases that don't expect this method to throw?

---

From slack: the impact of a bindings change is multi-version and multi-codepath and released drivers won't have handling to throw the appropriate error. 

I added handling for an error status in the collection info loop and we do have access to the error message via status.

What kind of errors could be encountered here?